modules/Bio/EnsEMBL/Analysis/Hive/Config/StarScallopRnaseq.pm

```diff
@@ -0,0 +1,827 @@
+    else {
+      $tables .= $key . ' VARCHAR(50) NOT NULL,'
+    }
+  }
+  $tables .= ' KEY(SM), KEY(ID)';
+
+  return [
+    # inheriting database and hive tables' creation
+    @{ $self->SUPER::pipeline_create_commands },
+
+    $self->db_cmd( 'CREATE TABLE ' . $self->o('summary_csv_table') . " ($tables)" ),
+
+    $self->db_cmd( 'CREATE TABLE ' . $self->o('read_length_table') . ' (' .
+        'fastq varchar(50) NOT NULL,' .
+        'read_length int(50) NOT NULL,' .
+        'PRIMARY KEY (fastq))' ),
+
+    'mkdir -p ' . $self->o('rnaseq_dir'),
+    'mkdir -p ' . $self->o('genome_dumps'),
+  ];
+}
+
+sub pipeline_wide_parameters {
+  my ($self) = @_;
+
+  return {
+    %{ $self->SUPER::pipeline_wide_parameters },
+    skip_rnaseq          => $self->o('skip_rnaseq'),
+    wide_ensembl_release => $self->o('ensembl_release'),
+    use_genome_flatfile  => $self->o('use_genome_flatfile'),
+    genome_file          => $self->o('faidx_genome_file'),
+    }
+}
+
+=head2 create_header_line
+
+ Arg [1]    : Arrayref String, it will contains the values of 'file_columns'
+ Example    : create_header_line($self->o('file_columns');
+ Description: It will create a RG line using only the keys present in your csv file
+ Returntype : String representing the RG line in a BAM file
+ Exceptions : None
+
+
+=cut
+
+sub create_header_line {
+  my ($items) = shift;
+
+  my @read_tags = qw(ID SM DS CN DT FO KS LB PG PI PL PM PU);
+  my $read_line = '@RG';
+  foreach my $rt (@read_tags) {
+    $read_line .= "\t$rt:#$rt#" if ( grep( $rt eq $_, @$items ) );
+  }
+  return $read_line . "\n";
+}
+
+## See diagram for pipeline structure
+sub pipeline_analyses {
+  my ($self) = @_;
+
+  my %commandline_params = (
+    'ncbi'        => '-num_threads 3 -window_size 40',
+    'wu'          => '-cpus 3 -hitdist 40',
+    'legacy_ncbi' => '-a 3 -A 40',
+  );
+  my $header_line = create_header_line( $self->default_options->{'file_columns'} );
+
+  return [
+
+############################################################################
+#
+# RNA-seq analyses
+#
+############################################################################
+    {
+      -logic_name => 'create_star_rnaseq_for_layer_db',
+      -module     => 'Bio::EnsEMBL::Analysis::Hive::RunnableDB::HiveCreateDatabase',
+      -parameters => {
+        source_db   => $self->o('dna_db'),
+        target_db   => $self->o('star_rnaseq_for_layer_db'),
+        create_type => 'clone',
+      },
+      -rc_name   => 'default',
+      -flow_into => {
+        1 => ['fan_star_rnaseq_for_layer_db'],
+      },
+    },
+
+    {
+      -logic_name => 'fan_star_rnaseq_for_layer_db',
+      -module     => 'Bio::EnsEMBL::Hive::RunnableDB::SystemCmd',
+      -parameters => {
+        cmd => 'if [ "#skip_rnaseq#" -ne "0" ]; then exit 42; else exit 0;fi',
+        return_codes_2_branches => { '42' => 2 },
+      },
+      -rc_name   => 'default',
+      -flow_into => {
+        1 => ['checking_file_path'],
+      },
+    },
+
```

Comment:
This can be removed as it will be in the main pipeline. The flow will need to be updated

Reply:
Removed "fan_star_rnaseq_for_layer_db" and updated flow in "create_star_rnaseq_for_layer_db".